artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/BrokerBalancerControlImpl.java

```diff
@@ -59,11 +59,10 @@ public BrokerBalancerControlImpl(final BrokerBalancer balancer, final StorageMan
 
    @Override
    public CompositeData getTarget(String key) throws Exception {
-      Target target = balancer.getTarget(key);
-
-      if (target != null) {
+      Target.Result result = balancer.getTarget(key);
+      if (Target.Status.OK.compareTo(result.status) == 0) {
          CompositeData connectorData = null;
-         TransportConfiguration connector = target.getConnector();
+         TransportConfiguration connector = result.target.getConnector();
```

Comment:
It looks a bit weird to me `Target.Result result...` and `result.target...`, why an inner class Target.Result and not another class TragetResult that includes Target and Status ?

Reply:
Perhaps similar wondering about a TargetResult class when considering the 'context.getResult()' usage in AMQPRedirectHandler . It would be nice if the status checking added using that was being done against the actual [Target.]Status enum, rather than needing to do object equality check against a full not-really-Target object. You could instead have a TargetResult holder indicate the result with the enum (and/or boolean conversion method checking for success) and not have any actual Target object.

Reply:
agree, I will replace that inner class with a separate class